CARDS-1631: Mark visits with no surveys as "No surveys assigned"

acrowthe opened the pull request:
- Add `has_surveys` question to visit information form
- Mark visits with no surveys by setting `has_surveys` to false, rather than setting `surveys_complete` to true
- Update VisitView component with `No surveys assigned` text

![image](https://user-images.githubusercontent.com/20056751/157334171-a2b898a4-d209-450b-9009-199c7f924023.png)

acrowthe committed:
CARDS-1631: Mark visits with no surveys as "No surveys assigned"
- Add `has_surveys` question to visit information form
- Mark visits with no surveys by setting `has_surveys` to false,
  rather than setting `surveys_complete` to true
- Update VisitView component with `No surveys assigned` text

marta- commented on proms-resources/backend/src/main/java/io/uhndata/cards/proms/internal/VisitChangeListener.java:
```diff
@@ -549,29 +554,32 @@ private boolean isFormIncomplete(final Node form)
+     * @param questionPath the relative path from the visit information questionnaire to the question to be answered
      */
-    private void changeVisitComplete(final Node visitInformationForm, final boolean complete, final Session session)
+    private void changeVisitInformation(final Node visitInformationForm, final boolean value, final Session session,
+        final String questionPath)
```
Minor code style comment: I would make `questionPath` the second parameter  to make the signature easier to read: ("change this `visitInformationForm`, more specifically this `questionPath`, to this `value`, using this `session`")

acrowthe committed:
CARDS-1631: Mark visits with no surveys as "No surveys assigned"
- Minor code formating refactor

marta- reviewed (approved):
Code looks good to me

sashaandjic commented:
Once I logged as patient and accepted ToU, I ended up with blank screen.
![image](https://user-images.githubusercontent.com/23243244/157503171-2f391375-5c78-4f61-852a-80b9d077e412.png)

sashaandjic commented:
Also, for a patient that haven't started filling in their surveys, the status is displayed as "No surveys assigned"
![image](https://user-images.githubusercontent.com/23243244/157506127-fcec2d49-dc46-4150-9a36-d24e41e83dd6.png)
![image](https://user-images.githubusercontent.com/23243244/157506387-d2a8d42d-ac61-4bc6-a40a-c83565032dbd.png)

acrowthe commented:
> Once I logged as patient and accepted ToU, I ended up with blank screen.

Issue in dev, not with this PR as discussed in standup. Urgent, but to be resolved elsewhere.

> Also, for a patient that haven't started filling in their surveys, the status is displayed as "No surveys assigned"

Issue reproduced. It's due to `has_surveys` only being automatically created the first time the form is saved. Since the normal editing process for forms saves twice (once when you click save, once when you navigate away from the page), `has_surveys` gets created with a null answer (first save), filled out by the backend, then overwritten to null by the second save (when closing the page)

acrowthe committed:
CARDS-1631: Mark visits with no surveys as "No surveys assigned"
- Fix issue where `has_surveys` was getting deleted by the front end and
  never recreated, leading to all forms being marked as "no surveys assigned"

sashaandjic commented:
I am still getting "No surveys assigned" for the visit that has surveys forms generated.
Steps to reproduce:
- create patient/visit form for tomorrow
-5 survey forms are created as expected
-in the visit form mark surveys as completed and submitted; change the date to Jan 1, 2022
- go to PMCC-ACHD and observe that the form has `Completed' in the Survey completion column
- create a new visit form for tomorrow
- 4 survey forms are crated
- go to PMCC-ACHD; the new visit form has `No survey assigned` displayed in the Survey completion column

acrowthe commented:
>  the new visit has No survey assigned displayed in the Survey completion column

Attempting to reproduce, when following the described steps I saw the new visit correctly marked as incomplete

> No surveys assigned will remain even if I mark the visit as completed or if I add a new survey manually

This is expected behavior per specs as `has_surveys` is only updated by the automatic form creation and takes priority over completed, though when 1691 is implemented it will no occur

acrowthe commented:
Rebased on top of dev

acrowthe committed:
CARDS-1631: Mark visits with no surveys as "No surveys assigned"
- Fix another case where `has_surveys` is not created. `has_surveys` should
  now be created any time a visit has `surveys` and `time` filled out

sashaandjic reviewed (commented):
tested - visits with no surveys generated are correctly marked as "No surveys assigned".
I found a bug (out of scope) with survey creation when they are not supposed to be created. 
In the attached screenshot the second visit should not have been assigned any surveys. 
I will log this as a separate task.
![image](https://user-images.githubusercontent.com/23243244/158006289-c28b584d-8d87-4f54-9554-d3391d7250d4.png)
